convert_hf_to_gguf.py

```diff
@@ -2160,6 +2163,47 @@ def set_vocab(self):
                     scores.append(-1000.0)
                     toktypes.append(SentencePieceTokenTypes.USER_DEFINED)
 
+        chat_eos_token = '<|im_end|>'
+        chat_eos_token_id = None
+
+        tokenizer_config_file = self.dir_model / 'tokenizer_config.json'
+        if tokenizer_config_file.is_file():
+            with open(tokenizer_config_file, "r", encoding="utf-8") as f:
+                tokenizer_config_json = json.load(f)
+                added_tokens_decoder = tokenizer_config_json.get("added_tokens_decoder", {})
+                for token_id, foken_data in added_tokens_decoder.items():
+                    token_id = int(token_id)
+                    token = foken_data["content"]
+                    if token == chat_eos_token:
+                        chat_eos_token_id = token_id
+                    token = token.encode("utf-8")
+                    if toktypes[token_id] != SentencePieceTokenTypes.UNKNOWN:
+                        assert(tokens[token_id] == token)
+                    tokens[token_id] = token
+                    scores[token_id] = -1000.0
+                    toktypes[token_id] = SentencePieceTokenTypes.USER_DEFINED
+                    if foken_data.get("special"):
+                        toktypes[token_id] = SentencePieceTokenTypes.CONTROL
+
+        tokenizer_file = self.dir_model / 'tokenizer.json'
```

Comment:
Getting tokens *both* from a SentencePiece tokenizer ***and*** `tokenizer.json` seems a bit weird.

Should `self._set_vocab_gpt2()` first be tried and then fallback to the SentencePiece tokenizer, or is there a particular reason not to do that?

Reply:
hi, thanks for your review. Yes. I referred to code from Phi3 and `tokenizer.json` does not exist for [internlm2](https://huggingface.co/collections/internlm/internlm2-65b0ce04970888799707893c) models and [internlm2.5](https://huggingface.co/collections/internlm/internlm25-66853f32717072d17581bc13) models.

Reply:
> But we are not sure for fine-tuned models from internlm2

I see. In that case I think this is fine.